libtsuba/src/RDGPartHeader.cpp

```diff
@@ -425,18 +426,43 @@ tsuba::to_json(json& j, const tsuba::PartitionMetadata& pmd) {
 
 void
 tsuba::from_json(const json& j, tsuba::PartitionMetadata& pmd) {
+  // magic is the only required field
   uint32_t magic;
   j.at("magic").get_to(magic);
-  j.at("policy_id").get_to(pmd.policy_id_);
-  j.at("transposed").get_to(pmd.transposed_);
-  j.at("is_outgoing_edge_cut").get_to(pmd.is_outgoing_edge_cut_);
-  j.at("is_incoming_edge_cut").get_to(pmd.is_incoming_edge_cut_);
-  j.at("num_global_nodes").get_to(pmd.num_global_nodes_);
-  j.at("num_global_edges").get_to(pmd.num_global_edges_);
-  j.at("num_nodes").get_to(pmd.num_nodes_);
-  j.at("num_edges").get_to(pmd.num_edges_);
-  j.at("num_owned").get_to(pmd.num_owned_);
-  j.at("cartesian_grid").get_to(pmd.cartesian_grid_);
+
+  if (auto it = j.find("policy_id"); it != j.end()) {
+    it->get_to(pmd.policy_id_);
+  }
+  if (auto it = j.find("transposed"); it != j.end()) {
+    it->get_to(pmd.transposed_);
+  }
+  if (auto it = j.find("is_outgoing_edge_cut"); it != j.end()) {
+    it->get_to(pmd.is_outgoing_edge_cut_);
+  }
+  if (auto it = j.find("is_incoming_edge_cut"); it != j.end()) {
+    it->get_to(pmd.is_incoming_edge_cut_);
+  }
+  if (auto it = j.find("num_global_nodes"); it != j.end()) {
+    it->get_to(pmd.num_global_nodes_);
+  }
+  if (auto it = j.find("max_global_node_id"); it != j.end()) {
+    it->get_to(pmd.max_global_node_id_);
+  }
+  if (auto it = j.find("num_global_edges"); it != j.end()) {
+    it->get_to(pmd.num_global_edges_);
+  }
+  if (auto it = j.find("num_nodes"); it != j.end()) {
+    it->get_to(pmd.num_nodes_);
+  }
+  if (auto it = j.find("num_edges"); it != j.end()) {
+    it->get_to(pmd.num_edges_);
+  }
+  if (auto it = j.find("num_owned"); it != j.end()) {
+    it->get_to(pmd.num_owned_);
+  }
+  if (auto it = j.find("cartesian_grid"); it != j.end()) {
+    it->get_to(pmd.cartesian_grid_);
+  }
```

Comment:
Would prefer to leave these required. We can do `if (..find)...` for new fields with the aim of removing them all eventually when we have better migration tools.

Reply:
How does it look now?